odl/util/testutils.py

```diff
@@ -236,29 +236,53 @@ def simple_fixture(name, params, fmt=None):
             to `pytest.fixture`.
         params : sequence
             Values to be taken as parameters in the fixture. They are
-            used as ``params`` argument to `pytest.fixture`.
+            used as ``params`` argument to `_pytest.fixtures.fixture`.
+            Arguments wrapped in a ``pytest.skipif`` decorator are
+            unwrapped for the generation of the test IDs.
         fmt : str, optional
             Use this format string for the generation of the ``ids``.
-            For each value, the id string is generated as::
+            For each value, the id string is generated as ::
 
                 fmt.format(name=name, value=value)
 
             hence the format string must use ``{name}`` and ``{value}``.
+            Default format strings are:
 
-            Default: ``" {name} = '{value}' "`` for string parameters,
-            otherwise ``" {name} = {value} "``
+                - ``" {name} = '{value}' "`` for string parameters,
+                - ``" {name} = {value} "`` for other types.
         """
+        import _pytest
+
         if fmt is None:
-            try:
-                params[0] + ''
-            except TypeError:
-                # Not a string type
-                fmt = " {name} = {value} "
-            else:
-                # String type
-                fmt = " {name} = '{value}' "
-
-        ids = [fmt.format(name=name, value=value) for value in params]
+            fmt_str = " {name} = '{value}' "
+            fmt_default = " {name} = {value} "
+
+            ids = []
+            for p in params:
+                # TODO: other types of decorators?
```

Comment:
Nice Todo, perhaps make an issue and reference it here

Reply:
Not sure if it's really a TODO, therefore the question mark. Depends on whether we ever encounter another type of decorator that we may want to unwrap.